src/catalog/src/remote/manager.rs

```diff
@@ -332,6 +201,203 @@ impl RemoteCatalogManager {
+    catalog_name: &'a str,
+) -> Pin<Box<dyn Stream<Item = Result<SchemaKey>> + Send + 'a>> {
+    let schema_prefix = build_schema_prefix(catalog_name);
+    let mut schemas = backend.range(schema_prefix.as_bytes());
+
+    Box::pin(stream!({
+        while let Some(r) = schemas.next().await {
+            let Kv(k, _) = r?;
+            if !k.starts_with(schema_prefix.as_bytes()) {
+                debug!("Ignoring non-schema key: {}", String::from_utf8_lossy(&k));
+                continue;
+            }
+
+            let schema_key =
+                SchemaKey::parse(&String::from_utf8_lossy(&k)).context(InvalidCatalogValueSnafu)?;
+            yield Ok(schema_key)
+        }
+    }))
+}
+
+async fn initiate_schemas(
+    node_id: u64,
+    backend: KvBackendRef,
+    engine_manager: TableEngineManagerRef,
+    catalog_name: &str,
+    catalog: CatalogProviderRef,
+    max_table_id: &mut TableId,
+) -> Result<()> {
+    let mut schemas = iter_remote_schemas(&backend, catalog_name).await;
+    let mut joins = Vec::new();
+    while let Some(r) = schemas.next().await {
+        let SchemaKey {
+            catalog_name,
+            schema_name,
+            ..
+        } = r?;
+
+        info!("Found schema: {}.{}", catalog_name, schema_name);
+        let schema = match catalog.schema(&schema_name).await? {
+            None => {
+                let schema = new_schema_provider(
+                    node_id,
+                    backend.clone(),
+                    engine_manager.clone(),
+                    &catalog_name,
+                    &schema_name,
+                );
+                catalog
+                    .register_schema(schema_name.clone(), schema.clone())
+                    .await?;
+                info!("Registered schema: {}", &schema_name);
+                schema
+            }
+            Some(schema) => schema,
+        };
+
+        info!(
+            "Fetch schema from metasrv: {}.{}",
+            &catalog_name, &schema_name
+        );
+        increment_gauge!(
+            crate::metrics::METRIC_CATALOG_MANAGER_SCHEMA_COUNT,
+            1.0,
+            &[crate::metrics::db_label(&catalog_name, &schema_name)],
+        );
+
+        let backend = backend.clone();
+        let engine_manager = engine_manager.clone();
+
+        joins.push(common_runtime::spawn_bg(async move {
+            let mut max_table_id = MAX_SYS_TABLE_ID;
+            initiate_tables(
+                node_id,
+                backend,
+                engine_manager,
+                &catalog_name,
+                &schema_name,
+                schema,
+                &mut max_table_id,
+            )
+            .await?;
+            Ok(max_table_id) as Result<TableId>
+        }));
+    }
+
+    let join_res = futures::future::join_all(joins).await;
+    for item in join_res {
+        let table_id = item.context(ParallelOpenTableSnafu)??;
+        *max_table_id = (*max_table_id).max(table_id);
+    }
```

Comment:
```suggestion
    if let Some(found_max_table_id) = futures::future::try_join_all(joins)
        .await
        .context(ParallelOpenTableSnafu)?
        .into_iter()
        .collect::<Result<Vec<_>>>()?
        .into_iter()
        .max()
    {
        *max_table_id = (*max_table_id).max(found_max_table_id);
    }
```

Reply:
Looks like it's not fixed, please close the conversation only when it's fixed.